pkg/generate/app/pipeline.go

```diff
@@ -43,7 +45,11 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		name = fmt.Sprintf("app-%04d", rand.Intn(10000))
+	}
+	if name == input.Name {
+		// Make output image name consistently different than input image name
+		name = namer.GetName(name, "app", v2.RepositoryNameTotalLengthMax)
```

Comment:
For the length you can always do 
```
v2.RepositoryNameTotalLengthMax - DNS1123LabelMaxLength - 1 
```
to adjust for the length of whatever the namespace is.